src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -581,7 +581,11 @@ export class RunActiveFileInTerminalAction extends Action {
 			this.notificationService.warn(nls.localize('workbench.action.terminal.runActiveFile.noFile', 'Only files on disk can be run in the terminal'));
 			return TPromise.as(void 0);
 		}
-		instance.sendText(uri.fsPath, true);
+
+		const filePath = this.terminalService.isWslBashTerminal(instance) ?
+			uri.path.replace(/([A-Za-z]):/, (m, g) => `mnt/${g.toLocaleLowerCase()}`) :
```

Comment:
I've been told that it's not safe to assume /mnt/x/ is the right path https://github.com/Tyriar/vscode-terminal-here/issues/15#issuecomment-366108775

There's also more information there about detecting the executable correctly.

Reply:
I'm stumped here. It [appears](https://docs.microsoft.com/en-us/windows/wsl/release-notes#build-17046) wslpath is available on build >= 17046. Is wslpath only available on insider builds? If so, can we rely on wslpath?